CRM/Membershipextrasimporterapi/EntityImporter/LineItem.php

```diff
@@ -0,0 +1,303 @@
+<?php
+
+use CRM_MembershipExtras_Service_MoneyUtilities as MoneyUtilities;
+
+class CRM_Membershipextrasimporterapi_EntityImporter_LineItem {
+
+  private $rowData;
+
+  private $contributionId;
+
+  private $membershipId;
+
+  private $cachedValues;
+
+  private $entityTable;
+
+  private $entityId;
+
+  private $contribution;
+
+  private $membership;
+
+  private $priceFieldValue;
+
+  public function __construct($rowData, $contributionId, $membershipId) {
+    $this->rowData = $rowData;
+    $this->contributionId = $contributionId;
+    $this->membershipId = $membershipId;
+    $this->setContribution($contributionId);
+    $this->setMembership($membershipId);
+    $this->setEntityTable();
+    $this->setEntityId();
+  }
+
+  private function setContribution($contributionId) {
+    $dao = CRM_Core_DAO::executeQuery("SELECT * FROM civicrm_contribution WHERE id = {$contributionId}");
```

Comment:
@omarabuhussein  I am wondering why we do not get from API?  Is it faster to query directly from SQL or any other reason for this? This applies for setMembership() too asd well as other entity where you can query from CiviCRM APIs.

Reply:
all the extension uses SQL queries (except in tests) for performance reasons,  Civi APIs are usually very slow and we want to be able to import large CSV files in a relatively short time, and thus why SQL queries are used.